From 9868d403221bc9d566cb88a37510a182b7fbad3b Mon Sep 17 00:00:00 2001
From: Daming Yang <lion@aosc.io>
Date: Tue, 17 Jul 2018 10:29:56 +0800
Subject: [PATCH] deps: backport crypto/init.c fix for openssl110

The statically linked program should not try to keep loaded
by dlopen trick.

Fixes: https://github.com/nodejs/node/issues/21845
Refs: https://github.com/openssl/openssl/pull/6725
---
 deps/openssl/openssl/crypto/init.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/deps/openssl/openssl/crypto/init.c b/deps/openssl/openssl/crypto/init.c
index 2d16c41bc60..8473b2a5831 100644
--- a/deps/openssl/openssl/crypto/init.c
+++ b/deps/openssl/openssl/crypto/init.c
@@ -91,7 +91,8 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_base)
      */
     base_inited = 1;
 
-#if !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
+#if defined(OPENSSL_NO_STATIC_ENGINE) && \
+    !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
 # ifdef DSO_WIN32
     {
         HMODULE handle = NULL;
@@ -621,7 +622,8 @@ int OPENSSL_atexit(void (*handler)(void))
 {
     OPENSSL_INIT_STOP *newhand;
 
-#if !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
+#if defined(OPENSSL_NO_STATIC_ENGINE) && \
+    !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
     {
         union {
             void *sym;
